File: app-core/src/main/java/code/sdk/core/util/AbstractPreference.java
package code.sdk.core.util;

import android.content.Context;
import android.content.SharedPreferences;

import code.util.AppGlobal;

public class AbstractPreference {

    /* private */
    protected static SharedPreferences getPreferences() {
        Context context = AppGlobal.getApplication();
        SharedPreferences preferences = context.getSharedPreferences("preference", Context.MODE_PRIVATE);
        //ObfuscationStub6.inject();
        return preferences;
    }

    protected static boolean putString(String key, String value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        //ObfuscationStub7.inject();
        return editor.putString(key, value).commit();
    }

    protected static String getString(String key) {
        //ObfuscationStub8.inject();
        return getPreferences().getString(key, "");
    }


    protected static boolean putBoolean(String key, boolean value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        //ObfuscationStub1.inject();
        return editor.putBoolean(key, value).commit();
    }

    protected static boolean getBoolean(String key, boolean defaultValue) {
        //ObfuscationStub2.inject();
        return getPreferences().getBoolean(key, defaultValue);
    }

    protected static boolean hasKey(String key) {
        //ObfuscationStub0.inject();
        return getPreferences().contains(key);
    }

    protected static boolean removeKey(String key) {
        //ObfuscationStub3.inject();
        SharedPreferences.Editor editor = getPreferences().edit();
        return editor.remove(key).commit();
    }
    /* private */

}

File: lib-util/src/main/java/book/util/AbstractPreference.kt
package book.util

import android.content.Context
import android.content.SharedPreferences


open class AbstractPreference(fileName: String) : IPreference {

    var preferences: SharedPreferences? = null

    init {
        preferences = AppGlobal.application?.getSharedPreferences(fileName, Context.MODE_PRIVATE)
    }

    override fun putString(key: String, value: String?): Boolean {
        if (preferences == null) return false
        return preferences!!.edit().putString(key, value).commit()
    }

    override fun getString(key: String): String? {
        if (preferences == null) return ""
        return getString(key, "")
    }

    override fun getString(key: String, defaultValue: String?): String? {
        if (preferences == null) return defaultValue
        return preferences!!.getString(key, defaultValue)
    }

    override fun putBoolean(key: String, value: Boolean): Boolean {
        if (preferences == null) return false
        return preferences!!.edit().putBoolean(key, value).commit()
    }

    override fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        if (preferences == null) return defaultValue
        return preferences!!.getBoolean(key, defaultValue)
    }

    override fun putLong(key: String, value: Long): Boolean {
        if (preferences == null) return false
        return preferences!!.edit().putLong(key, value).commit()
    }

    override fun getLong(key: String, defaultValue: Long): Long {
        if (preferences == null) return defaultValue
        return preferences!!.getLong(key, defaultValue)
    }

    override fun putInt(key: String, value: Int): Boolean {
        if (preferences == null) return false
        return preferences!!.edit().putInt(key, value).commit()
    }

    override fun getInt(key: String, defaultValue: Int): Int {
        if (preferences == null) return defaultValue
        return preferences!!.getInt(key, defaultValue)
    }

    override fun hasKey(key: String): Boolean {
        if (preferences == null) return false
        return preferences!!.contains(key)
    }

    override fun removeKey(key: String): Boolean {
        if (preferences == null) return false
        return preferences!!.edit().remove(key).commit()
    }

}